README.md

```diff
@@ -1 +1,431 @@
+| `CGFloat`    | `%f` or `%.nf`, *where* `n` *is the number of decimal places to display* |
+| `BOOL`       | `%d`             |
+
+Examples:
+
+```objc
+NSLog(@"%@", @"Welcome!");
+NSLog(@"-1: %li", -1);
+NSLog(@"42: %lu", 42);
+NSLog(@"pi: %f, 3.141593);
+NSLog(@"sqrt2: %.12f", 1.41421356237);
+NSLog(@"yes: %d", YES);
+```
+These will print:
+
+```
+Welcome!
+-1: -1
+42: 42
+pi: 3.141593
+sqrt2: 1.414213562370
+yes: 1
+```
+
+[Back to Top](#table-of-contents)
+## Primitives
+
+### Common Primitives
+
+| Primitive   | Description |
+|:-----------:|:------------|
+| `NSInteger` | An integer value that can be either positive or negative. |
+| `NSUInteger`| An integer value that can **only** be positive (the "U" means "unsigned"). |
+| `CGFloat`   | A "float" value that can (imperfectly) hold a decimal point. **Note:** *Do not use float values to hold currency.* |
+| `BOOL`      | A `YES` or `NO` value: `0` means "no", `1` means "yes" |
+
+### Declaring A Primitive Variable
+
+Syntax:
+
+```objc
+Type variableName = value;
+```
+
+Examples:
+
+```objc
+NSInteger i = 0;
+NSUInteger u = 0;
+CGFloat f = 0.0;
+BOOL hidden = NO;
+```
+**Reminder:** *Do not use a* `*` *when declaring primitives. Entering* `NSInteger *variable` *will cause the compiler to generate an* `invalid integer to pointer conversion` *warning. The same goes for declaring any of the data types. This syntax does serve a purpose so the language permits it, but your application won't do what you expect.*
+
+### Redefining A Primitive Variable
+
+Syntax:
+
+```objc
+variableName = value;
+```
+
+Examples:
+
+```objc
+i = -1;
+u = 1;
+f = 3.14159265359;
+hidden = YES;
+```
+
+### Printing A Primitive Variable
+
+See [Format Specifiers](#format-specifiers).
+
+Examples:
+
+```objc
+NSLog(@"i: %li", i);
+NSLog(@"u: %lu", u);
+NSLog(@"f: %f", f);
+NSLog(@"f: %.12f", f);
+NSLog(@"hidden: %d", hidden);
+```
+
+This will print:
+
+```
+i: -1
+u: 1
+f: 3.141593
+f: 3.141592653590
+hidden: 1           // 1 means 'yes'
+```
+[Back to Top](#table-of-contents)
+## Operators
+
+### Mathematical Operators
+
+| Operator | Name               | Pronunciation | Description |
+|:--------:|:-------------------|:--------------|:------------|
+| `+` | Addition Operator       | "plus" or "add"       | Results to the value that is the **sum** of the two operands. |
+| `-` | Subtraction Operator    | "minus" or "subtract" | Results to the **difference** of subtracting the right operand from the left operand.  |
+| `*` | Multiplication Operator | "star" or "times"     | Results to the **product** of multiplying the two operands. |
+| `/` | Division Operator       | "slash" or "over"     | Results to the **quotient** of dividing the left operand by the right operand. **Note:** *This operator truncates integer-only divisions.* |
+| `%` | **Advanced:** Modulus or Modulo       | "modulus" or "modulo" | Results to the **remainder** of clocking the left operand around the right operand. |
```

Comment:
that's interesting, i expected `-3 % 26` to return `23`, not `-3`. Maybe modulus works differently in other languages?